thrift/thrift-gen/tchannel-template.go

```diff
@@ -142,46 +179,61 @@ func (s *{{ .ServerStruct }}) Handle(ctx {{ contextType }}, methodName string, p
 }
 
 {{ range .Methods }}
-	func (s *{{ $svc.ServerStruct }}) {{ .HandleFunc }}(ctx {{ contextType }}, protocol athrift.TProtocol) (bool, athrift.TStruct, error) {
+	func (s *{{ $svc.ServerStruct }}) {{ .HandleFunc }}(ctx {{ contextType }}, protocol athrift.TProtocol) (handled bool, resp athrift.TStruct, err error) {
 		var req {{ .ArgsType }}
 		var res {{ .ResultType }}
+		serviceMethod := "{{ .ThriftName }}.{{ .Name }}"
 
-		if err := req.Read(protocol); err != nil {
+		defer func () {
```

Comment:
I think the handling of the panics should happen at the TChannel level, where it kicks off a goroutine. I have ideas for how to handle that, let's do that separately.